File: app/src/main/java/com/simplemobiletools/calculator/operation/RootOperation.java
package com.simplemobiletools.calculator.operation;

import com.simplemobiletools.calculator.operation.base.Operation;
import com.simplemobiletools.calculator.operation.base.UnaryOperation;

public class RootOperation extends UnaryOperation implements Operation {

    protected RootOperation(double value) {
        super(value);
    }

    @Override
    public double getResult() {
        return Math.sqrt(value);
    }
}

File: app/src/main/kotlin/com/simplemobiletools/calculator/operation/RootOperation.kt
package com.simplemobiletools.calculator.operation

import com.simplemobiletools.calculator.operation.base.BinaryOperation
import com.simplemobiletools.calculator.operation.base.Operation
import kotlin.math.sqrt

class RootOperation(baseValue: Double, secondValue: Double) : BinaryOperation(baseValue, secondValue), Operation {

    override fun getResult() = sqrt(secondValue) * baseValue
}
